amr-wind/physics/MultiRoughness.cpp

```diff
@@ -0,0 +1,94 @@
+#include "amr-wind/physics/MultiRoughness.H"
+#include "amr-wind/CFDSim.H"
+#include "AMReX_iMultiFab.H"
+#include "AMReX_MultiFabUtil.H"
+#include "AMReX_ParmParse.H"
+#include "AMReX_ParReduce.H"
+#include "amr-wind/utilities/trig_ops.H"
+#include "amr-wind/utilities/IOManager.H"
+
+namespace amr_wind::multiroughness {
+
+namespace {} // namespace
+
+MultiRoughness::MultiRoughness(CFDSim& sim)
+    : m_sim(sim)
+    , m_repo(sim.repo())
+    , m_mesh(sim.mesh())
+    , m_velocity(sim.repo().get_field("velocity"))
+    , m_terrainz0(sim.repo().declare_field("lowerz0", 1, 1, 1))
+{
+    std::string roughnessfile("ground.roughness");
+    std::ifstream file1(roughnessfile, std::ios::in);
+    if (!file1.good()) {
+        amrex::Abort("Cannot find ground.roughness");
```

Comment:
Change to: "Cannot find the file named ground.roughness". Also I would prefer not to use a `.` in there since that denotes an extension. `ground_roughness.csv` ? `ground_roughness.txt` ?